### PR TITLE
Add yast_nfs_client PM in yast command line mode

### PR DESCRIPTION
**Pull request by badboywj**
Regression test for Yast2 cmd nfs-client PM
- Related ticket: https://progress.opensuse.org/issues/49295
- Needles: no
- Verification run:
     - SLE12SP2: http://10.67.20.116/tests/638 :heavy_check_mark: 
     - SLE12SP3: http://10.67.20.116/tests/639 :heavy_check_mark: 
     - SLE12SP4: http://10.67.20.116/tests/640 :heavy_check_mark: 
     - SLE15      : http://10.67.20.116/tests/641 :heavy_check_mark: 
     - SLE15SP1: http://10.67.20.116/tests/642 :heavy_check_mark:

### CODE REVIEW

**Comment by badboywj** [comment]
@dzedro Could you help me review the code?

File: tests/yast2_cmd/yast_nfs_client.pm [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    assert_script_run("yast nfs edit spec=localhost:$servdir file=$tmpdir02");
+    validate_script_output('yast nfs list 2>&1', sub { m%$tmpdir02% }, timeout => 90, proceed_on_failure => 1);
+    validate_script_output('cat /proc/mounts', sub { m%$tmpdir02% and (!m%$tmpdir01%) }) if (is_sle('<15'));
+
+    assert_script_run("yast nfs delete spec=localhost:$servdir");
+    validate_script_output('yast nfs list 2>&1', sub { !m%$tmpdir02% }, timeout => 90, proceed_on_failure => 1);
+    validate_script_output('cat /proc/mounts', sub { !m%$tmpdir02% }) if (is_sle('<15'));
+
+    # restores all configs
+    record_soft_failure('bsc#1151426: stop nfs client service.') if (systemctl('is-active nfs.service', ignore_failure => 1));
```
Comment (dzedro) [inline tests/yast2_cmd/yast_nfs_client.pm]: As you can see in your test run, record_soft_failure will be executed when the nfs client is stopped as 0 is false in perl https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/tests/console/apache.pm#L38

File: tests/yast2_cmd/yast_nfs_client.pm [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    my $servdir = script_output('mktemp -d');
+    assert_script_run("yast nfs-server start; yast nfs-server add mountpoint=$servdir");
+
+    my $tmpdir01 = script_output('mktemp -d');
+    my $tmpdir02 = script_output('mktemp -d');
+
+    # adds,edits and deletes a mount point and verify
+    assert_script_run("yast nfs add spec=localhost:$servdir file=$tmpdir01");
+    validate_script_output('yast nfs list 2>&1', sub { m%$tmpdir01% }, timeout => 90, proceed_on_failure => 1);
+    # nfs.service will run "mount -at nfs,nfs4" to mount all points on SLE12SP2~SP4, but NOT on SLE15+
```
Comment (dzedro) [inline tests/yast2_cmd/yast_nfs_client.pm]: If on 15+ nfs.service does not mount nfs mounts then something else has to do it
There is no verification on 15+ without check of mounts

File: tests/yast2_cmd/yast_nfs_client.pm (hunk 1 above)
Comment (badboywj) [reply tests/yast2_cmd/yast_nfs_client.pm]: yes, it is clearer. I will change this line.

File: tests/yast2_cmd/yast_nfs_client.pm (hunk 2 above)
Comment (badboywj) [reply tests/yast2_cmd/yast_nfs_client.pm]: I had thought this point, and I thought "mount" action belong to "nfs.service" but not "yast nfs-client",  so I didn't verify it on SLE15+. Yes, there must be verifications on SLE15+, it is robust.

Comment (dzedro) [reply tests/yast2_cmd/yast_nfs_client.pm]: You could create function of the repeating code for verification

**Pull request merged**